packages/kbn-config/src/internal_mocks.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Observable, of } from 'rxjs';
+import { DocLinks } from '@kbn/doc-links';
+import { PublicMethodsOf } from '@kbn/utility-types';
+import type { EnvOptions } from './env';
+import type { RawConfigService } from './raw';
+import type { ConfigDeprecationContext } from './deprecation';
+
+type DeepPartial<T> = {
+  [P in keyof T]?: T[P] extends Array<infer R> ? Array<DeepPartial<R>> : DeepPartial<T[P]>;
+};
+
+export function getEnvOptions(options: DeepPartial<EnvOptions> = {}): EnvOptions {
+  return {
+    configs: options.configs || [],
+    cliArgs: {
+      dev: true,
+      silent: false,
+      watch: false,
```

Comment:
Due to the new package architecture, mocks used internally by a package's tests must be separated from the public mock, as the implementation package cannot import from the mock package.